### DISCUSSION TITLE
getting error with a file upload test

### DISCUSSION THREAD

**Discussion by vishytk** [spatie/laravel-activitylog#927] [Q&A]
I have following test that is failing with error:

Illuminate\Database\Eloquent\JsonEncodingException : Unable to encode attribute [properties] for model [Spatie\Activitylog\Models\Activity] to JSON: Type is not supported.

        $this->newUser = [
            'username' => 'newtestuser',
            'password' => 'secret',
            'firstname' => 'FirstName',
            'dob' => '04-05-2006',
            'role_id' => 1,
            'gender_id' => 1,
        ];

    public function a_manager_can_create_a_user_with_avatar(): void
    {
        $this->withoutExceptionHandling();

        $this->newUser['avatar'] = UploadedFile::fake()->image('avatar.jpg');

        $response = $this->actingAs($this->manager)
            ->postJson('/api/users', $this->newUser, $this->headers);
        dump($response);

        $createdUser = User::where('username', 'newtestuser')->first();

        $response->assertStatus(201)
            ->assertJsonFragment([
                'message' => 'user successfully created',
            ]);

        $this->assertEquals('newtestuser', $createdUser->username);
        $this->assertEquals('FirstName', $createdUser->firstname);
        $this->assertEquals('1', $createdUser->role_id);
        $this->assertEquals('1', $createdUser->gender_id);

        $storage_path = storage_path('app/public/images/user_avatars');
        $avatar_path = $storage_path . '/' . $createdUser->id . '/' . $createdUser->avatar;
        $this->assertFileExists($avatar_path);
    }

There are two other tests one creating a user without avatar and other updating avatar which are identical to this test with appropriate changes and both are working. 

What is the issue here ?

**Comment by Gummibeer** [comment]
I assume it's `$this->newUser['avatar'] = UploadedFile::fake()->image('avatar.jpg');` I have no idea how eloquent or your controller handle that `avatar` attribute. But I assume that an instance of `UploadFile` is passed to the array used for JSON generation instead of the base64 or the filepath string.

**Reply by vishytk** [reply]
At first, I also thought so. But the following test is working

    public function a_manager_can_update_a_users_profile_picture(): void
    {
        $this->withoutExceptionHandling();

        $user = UserFactory::new()->create();

        $avatar_picture = [
            'avatar' => UploadedFile::fake()->image('avatar.jpg'),
        ];
        $response = $this->actingAs($this->manager)
            ->patchJson('/api/users/' . $user->id, $avatar_picture, $this->headers);

        $createdUser = User::where('username', $user->username)->first();

        $response->assertStatus(200);

        $storage_path = storage_path('app/public/images/user_avatars');

        $avatar_path = $storage_path . '/' . $createdUser->id . '/' . $createdUser->avatar;

        $this->assertFileExists($avatar_path);
    }

**Reply by vishytk** [reply]
Controller method

 public function store(StoreUserRequest $request): JsonResponse
    {
        if ($request->hasFile('avatar')) {
            $user = $this->storeWithAvatar(User::create(request()->all()));
        } else {
            $user = User::create(request()->all());
        }

        return response()->json([
            'message' => 'user successfully created',
            'data' => $user,
        ], 201);
    }
